src/runner/math-patches.js

```diff
@@ -0,0 +1,97 @@
+const DEC_128 = "decimal128";
+const BIG_DECIMAL = "bigdec";
+
+const createUnaryHandler = (substituteFns) => ({
+  apply(target, thisArg, argsList) {
+    const [arg] = argsList;
+    if (arg instanceof Decimal) {
+      return substituteFns[DEC_128](arg);
+    }
+
+    if (arg instanceof Big) {
+      return substituteFns[BIG_DECIMAL](arg);
+    }
```

Comment:
Repeating the instance checks is kinda gross, BUT it means we only have to patch once and don't need to track the state.

Reply:
Not sure I'm a fan of this, seems like a lot of hassle to avoid tracking the one piece of state that is fundamental to the application :smile: It also means we have to import both Big and Decimal in the tests even if we don't use one of them, or else write `if (Big && arg instanceof Big)`. I'm fine to go with it for now, but I think it'll make things more complicated in the thing I plan to do next, which is implementing `Decimal.round()`. But I can cross that bridge when I come to it.